tt_metal/hw/firmware/src/erisc.cc

```diff
@@ -28,13 +28,50 @@ uint32_t wIndex __attribute__((used));
 
 tt_l1_ptr mailboxes_t * const mailboxes = (tt_l1_ptr mailboxes_t *)(eth_l1_mem::address_map::ERISC_MEM_MAILBOX_BASE);
 
+uint8_t noc_index = 0;  // TODO: remove hardcoding
 uint8_t my_x[NUM_NOCS] __attribute__((used));
 uint8_t my_y[NUM_NOCS] __attribute__((used));
 
 uint32_t noc_reads_num_issued[NUM_NOCS] __attribute__((used));
 uint32_t noc_nonposted_writes_num_issued[NUM_NOCS] __attribute__((used));
 uint32_t noc_nonposted_writes_acked[NUM_NOCS] __attribute__((used));
 
+FORCE_INLINE
+void multicore_eth_cb_wait_front(db_cb_config_t *eth_db_cb_config, int32_t num_pages) {
```

Comment:
I might modify this to create multicore cb apis that work for reader/writer being different core types. since we need this sync mechanism for multiple dispatch cores

Reply:
I think we need `_eth_` variants of every noc stall, because we need to call a context switch in the while loops if we want DPRINT to flush for remote chips during debug.

Reply:
I just haven't added the context switch here yet